packages/jest-circus/src/index.ts

```diff
@@ -1,206 +1,18 @@
-) => {
-  const asyncError = new ErrorWithStack(undefined, hookFn);
-
-  if (typeof fn !== 'function') {
-    asyncError.message =
-      'Invalid first argument. It must be a callback function.';
-
-    throw asyncError;
-  }
-
-  dispatch({asyncError, fn, hookType, name: 'add_hook', timeout});
-};
-
-// Hooks have to pass themselves to the HOF in order for us to trim stack traces.
-const beforeEach: THook = (fn, timeout) =>
-  _addHook(fn, 'beforeEach', beforeEach, timeout);
-const beforeAll: THook = (fn, timeout) =>
-  _addHook(fn, 'beforeAll', beforeAll, timeout);
-const afterEach: THook = (fn, timeout) =>
-  _addHook(fn, 'afterEach', afterEach, timeout);
-const afterAll: THook = (fn, timeout) =>
-  _addHook(fn, 'afterAll', afterAll, timeout);
-
-const test: Global.It = (() => {
-  const test = (testName: TestName, fn: TestFn, timeout?: number): void =>
-    _addTest(testName, undefined, fn, test, timeout);
-  const skip = (testName: TestName, fn?: TestFn, timeout?: number): void =>
-    _addTest(testName, 'skip', fn, skip, timeout);
-  const only = (testName: TestName, fn: TestFn, timeout?: number): void =>
-    _addTest(testName, 'only', fn, test.only, timeout);
-
-  test.todo = (testName: TestName, ...rest: Array<any>): void => {
-    if (rest.length > 0 || typeof testName !== 'string') {
-      throw new ErrorWithStack(
-        'Todo must be called with only a description.',
-        test.todo,
-      );
-    }
-    return _addTest(testName, 'todo', () => {}, test.todo);
-  };
-
-  const _addTest = (
-    testName: TestName,
-    mode: TestMode,
-    fn: TestFn | undefined,
-    testFn: (testName: TestName, fn: TestFn, timeout?: number) => void,
-    timeout?: number,
-  ) => {
-    const asyncError = new ErrorWithStack(undefined, testFn);
-
-    if (typeof testName !== 'string') {
-      asyncError.message = `Invalid first argument, ${testName}. It must be a string.`;
-
-      throw asyncError;
-    }
-    if (fn === undefined) {
-      asyncError.message =
-        'Missing second argument. It must be a callback function. Perhaps you want to use `test.todo` for a test placeholder.';
-
-      throw asyncError;
-    }
-    if (typeof fn !== 'function') {
-      asyncError.message = `Invalid second argument, ${fn}. It must be a callback function.`;
-
-      throw asyncError;
-    }
-
-    return dispatch({
-      asyncError,
-      fn,
-      mode,
-      name: 'add_test',
-      testName,
-      timeout,
-    });
-  };
-
-  test.each = bindEach(test);
-  only.each = bindEach(only);
-  skip.each = bindEach(skip);
-
-  test.only = only;
-  test.skip = skip;
-
-  return test;
-})();
-
-const it: Global.It = test;
-
-export = {
+export {addEventHandler, getState} from './state';
+export {EventHandler, Event, State} from './types';
+export {
   afterAll,
   afterEach,
   beforeAll,
   beforeEach,
   describe,
-  it,
   test,
-};
+  it,
+} from './globals';
+
+// To avoid breaking change. Remove in Jest 25:
+export default globals;
```

Comment:
I still think this is breaking since it breaks compat with `const circus = require('jest-circus');` (they'll have to do `require('jest-circus').default`). Not sure how big of a deal that is, though.

You can stick every export into `export = {}`, then we can fix it properly for jest 25? We need to ditch `export =` anyways (see https://github.com/DefinitelyTyped/DefinitelyTyped/pull/33705#issuecomment-473034112)

Reply:
I’ll triple check importing it into a common JS project but I’m fairly certain I’ve set it up to work same as before with ES6 and without.

Reply:
`export = {}` turns into `module.exports = {}`. `export default {}` turns into `module.exports.default = {}`

Note that `export =` is TS specific syntax for CJS compat, it's not part of the ESM spec

Reply:
Yup. That covers importing it as default into ES6.

But I’m ALSO exporting each one directly. it, describe, beforeEach, etc are all exported individually as well.

Reply:
Oh, I missed that! Then I guess we're good 😀 
Easiest way to check is just do `node -p 'require("jest-circus")'` from your branch after running `yarn build`.